bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/ui/view/visualizer/Visualizer.java

```diff
@@ -680,57 +681,57 @@ private void paintGroupBars(Graphics2D g2d) {
 	 * optionally as a closed shape corresponding to the area under the curve. These shapes
 	 * are cached.
 	 */
-	private Shape getElevationCurve(WorldCoords coords, long start, long end, boolean close, Map<WorldCoords, CachedShape> cache) {
+	private Shape getElevationCurve(Function<Long, WorldCoords> coords, long start, long end, boolean close, Map<WorldCoords, CachedShape> cache) {
 		synchronized (cacheLock) {
 
 			// Return the cached shape, if any.
-			CachedShape cs = cache.get(coords);
-			if (cs != null && cs.matches(start, end))
-				return cs.shape;
+//			CachedShape cs = cache.get(coords);
+//			if (cs != null && cs.matches(start, end))
+//				return cs.shape;
```

Comment:
Another idea would be to use the observation id as the cache key.  It's true that two or more observations could have the same target and therefore the cache might not be as efficient as it otherwise would be, but at least it would be guaranteed correct.